### PR TITLE
MISUV-1870 Moved agent-client relationship check to auth functionality.

### PR DESCRIPTION
**Pull request by jameshforster**

### CODE REVIEW

**Review by MihailsC-hmrc — CHANGES_REQUESTED** [review]
Please align the code lines that are indented too far. Other comments are optional.

File: app/auth/BaseFrontendController.scala [hunk 1]
```diff
@@ -47,17 +49,27 @@ abstract class BaseFrontendController(implicit val mcc: MessagesControllerCompon
 
     def apply(action: Request[AnyContent] => User => Result): Action[AnyContent] = async(action andThen (_ andThen Future.successful))
 
-    protected def asyncInternal(predicate: AuthPredicate[User])(action: ActionBody[User]): Action[AnyContent] =
+    protected def asyncInternal(predicate: AuthPredicate[User], requireClientSelected: Boolean = true)(action: ActionBody[User]): Action[AnyContent] =
```
[the remark this answers is not in the document]
Comment (jameshforster) [reply app/auth/BaseFrontendController.scala:52]: Action is already not called unless auth passes as it requires the user to be passed in as a variable before execution

File: app/controllers/predicates/IncomeTaxAgentUser.scala [hunk 2]
```diff
@@ -30,6 +30,9 @@ case class IncomeTaxAgentUser(enrolments: Enrolments,
                               confidenceLevel: ConfidenceLevel) extends IncomeTaxUser {
 
   lazy val agentReferenceNumber: Option[String] = getEnrolment(Constants.agentServiceEnrolmentName)
+	lazy val delegatedMtdEnrolments: Set[Enrolment] = enrolments.enrolments.filter(enrolment => {
```
[the remark this answers is not in the document]
Comment (jameshforster) [reply app/controllers/predicates/IncomeTaxAgentUser.scala:33]: Removed in later subtask

File: app/services/agent/ClientRelationshipService.scala [hunk 3]
```diff
@@ -32,15 +33,12 @@ class ClientRelationshipService @Inject()(agentClientRelationshipsConnector: Age
                                           incomeTaxViewChangeConnector: IncomeTaxViewChangeConnector)
                                          (implicit ec: ExecutionContext) {
 
-  def checkAgentClientRelationship(utr: String, arn: String)(implicit hc: HeaderCarrier): Future[Either[ClientRelationshipFailure, ClientDetails]] =
+  def checkAgentClientRelationship(utr: String, arn: String, delegatedEnrolments: Set[Enrolment])(implicit hc: HeaderCarrier): Future[Either[ClientRelationshipFailure, ClientDetails]] =
```
[the remark this answers is not in the document]
Comment (jameshforster) [reply app/services/agent/ClientRelationshipService.scala:36]: Removed in later subtask

File: app/services/agent/ClientRelationshipService.scala [hunk 4]
```diff
@@ -32,15 +33,12 @@ class ClientRelationshipService @Inject()(agentClientRelationshipsConnector: Age
...
     citizenDetailsConnector.getCitizenDetailsBySaUtr(utr) flatMap {
       case CitizenDetailsModel(optionalFirstName, optionalLastName, Some(nino)) =>
         incomeTaxViewChangeConnector.getBusinessDetails(nino) flatMap {
           case IncomeSourceDetailsModel(mtdbsa, _, _, _) =>
-            agentClientRelationshipsConnector.agentClientRelationship(arn, mtdbsa) map {
-              case true => Right(ClientRelationshipService.ClientDetails(optionalFirstName, optionalLastName, nino, mtdbsa))
-              case false => Left(NoAgentClientRelationship)
```
[the remark this answers is not in the document]
Comment (jameshforster) [reply app/services/agent/ClientRelationshipService.scala:42]: Removed in cleanup subtask

**Pull request merged**